### DISCUSSION TITLE
请问win32Controller如何实现只移动鼠标指针，不触发click动作呢？

### DISCUSSION THREAD

**Discussion by kkkkninezzz** [MaaXYZ/MaaFramework#323] [Q&A]
尝试使用过swipe，会触发一次click，但我想要实现的是在游戏里移动视角，请问有什么方式可以实现吗

**Comment by MistEO** [comment]
可以试试自定义动作用 touch move，2.0 版本或许会在 Pipeline 里也支持下

**Reply by kkkkninezzz** [reply]
看了touch move的api
```python
def touch_move(self, contact: int, x: int, y: int, pressure: int) -> bool:
        """
        Sync context touch move.

        :param contact: contact
        :param x: x
        :param y: y
        :param pressure: pressure

        :return: True if success, otherwise False
        """
        return bool(
            Library.framework.MaaSyncContextTouchMove(
                self._handle, contact, x, y, pressure
            )
        )
```
没太理解每个参数的定义，我现在是想实现在pc端鸣潮游戏里，调整玩家视角，似乎通过touch move无法实现？
我对win32 api不太了解，请问我怎么去实现这个功能呢

**Reply by weinibuliu** [reply]
~~touch 系列函数需要配合 touch_down(按下) 与 touch_move(抬起) 一起使用~~

touch_move api 中的 contact 是指“手指”（“光标”）序号，用来实现多指触控（在 Win32 中或许没有实质意义？）
x,y 是指目前“手指”（“光标”）的坐标
pressure 是指“压力”（应该特指安卓端中按压的像素点个数）

区别于 swipe 传入起点坐标与终点坐标，touch_move 在实际使用中，需要多次传入新的 x,y 值，以实现“滑动”。

**Reply by MistEO** [reply]
> touch 系列函数需要配合 touch_down(按下) 与 touch_move(抬起) 一起使用
> 
> touch_move api 中的 contact 是指“手指”（“光标”）序号，用来实现多指触控（在 Win32 中或许没有实质意义？） x,y 是指目前“手指”（“光标”）的坐标 pressure 是指“压力”（应该特指安卓端中按压的像素点个数）
> 
> 区别于 swipe 传入起点坐标与终点坐标，touch_move 在实际使用中，需要多次传入新的 x,y 值，以实现“滑动”。

不需要，touch move 就是移动鼠标，和按下抬起没有关系

**Comment by MistEO** [comment]
5.0 版本可以再试试
